corehq/apps/accounting/models.py

```diff
@@ -1630,7 +1630,11 @@ def renew_subscription(self, note=None, web_user=None,
         # record renewal from old subscription
         SubscriptionAdjustment.record_adjustment(
             self, method=adjustment_method, note=note, web_user=web_user,
-            reason=SubscriptionAdjustmentReason.RENEW,
+            reason=SubscriptionAdjustmentReason.RENEW, related_subscription=renewed_subscription
+        )
+        SubscriptionAdjustment.record_adjustment(
+            renewed_subscription, method=adjustment_method, note=note, web_user=web_user,
+            reason=SubscriptionAdjustmentReason.CREATE
```

Comment:
Should these two `record_adjustment` calls be grouped under a single transaction? It would seem possible that we could lose connectivity between the two calls, and then wind up with an inconsistent history, right?

Reply:
I don't think it's necessary. The two subscription adjustments are conceptually related but don't depend on each other, and I also don't think we would want block the subscription renewal itself on being able to record the subscription adjustments (if we used `@transaction.atomic` on the method, like with `change_plan`).

Reply:
Offlined and came to the opposite conclusion from my previous comment - we do want to make sure the adjustments are always recorded, it's important to have accurate records since we use them for investigation/debugging. https://github.com/dimagi/commcare-hq/pull/36979/commits/fa0f816f365f6d2cb95566745ffb8975282eab00